interop/test_utils.go

```diff
@@ -684,98 +685,244 @@ func DoPickFirstUnary(ctx context.Context, tc testgrpc.TestServiceClient) {
 	}
 }
 
-func doOneSoakIteration(ctx context.Context, tc testgrpc.TestServiceClient, resetChannel bool, serverAddr string, soakRequestSize int, soakResponseSize int, dopts []grpc.DialOption, copts []grpc.CallOption) (latency time.Duration, err error) {
-	start := time.Now()
-	client := tc
-	if resetChannel {
-		var conn *grpc.ClientConn
-		conn, err = grpc.Dial(serverAddr, dopts...)
+// SoakIterationResult represents the result of a single iteration in the soak test.
+type SoakIterationResult struct {
+	LatencyMs int64
+	Status    string // The status of the iteration
+}
+
+// ThreadResults stores the aggregated results for a specific thread during the soak test
+type ThreadResults struct {
+	IterationsDone int
+	Failures       int
+	Latencies      *stats.Histogram
+}
+
+// ChannelFunc can be used to customize how the channel is handled (reuse or reset)
+type ChannelFunc func(*grpc.ClientConn) (*grpc.ClientConn, testgrpc.TestServiceClient)
+
+// createChannel Initialize the shared channel (once for all threads)
+func createChannel(serverAddr string, dopts []grpc.DialOption) (*grpc.ClientConn, testgrpc.TestServiceClient) {
+	conn, err := grpc.NewClient(serverAddr, dopts...)
+	if err != nil {
+		log.Fatalf("Failed to create shared channel: %v", err)
+	}
+	client := testgrpc.NewTestServiceClient(conn)
+	return conn, client
+}
+
+func closeChannel(channel *grpc.ClientConn) {
+	if channel != nil {
+		err := channel.Close()
 		if err != nil {
-			return
+			log.Fatalf("Failed to close channel: %v", err)
 		}
-		defer conn.Close()
-		client = testgrpc.NewTestServiceClient(conn)
 	}
-	// per test spec, don't include channel shutdown in latency measurement
-	defer func() { latency = time.Since(start) }()
-	// do a large-unary RPC
+}
+
+// CreateNewChannel Create a new channel by shutting down the current one (for channel soak tests)
```

Comment:
Nit: The comments for functions (here and elsewhere) should be of the form: `// CreateFoo creates a new Foo by bar baz.`.